libnavui-speedlimit/src/main/java/com/mapbox/navigation/ui/speedlimit/view/MapboxSpeedInfoView.kt

```diff
@@ -0,0 +1,448 @@
+                speedInfoViennaLayout.isVisible = false
+            }
+        }
+    }
+
+    /**
+     * Apply runtime changes to [MapboxSpeedLimitView] using [MapboxSpeedInfoOptions]
+     * @param speedInfoOptions
+     */
+    fun applyOptions(speedInfoOptions: MapboxSpeedInfoOptions) {
+        this.speedInfoOptions = speedInfoOptions
+        updateView()
+    }
+
+    private fun showMutcdOrVienna(speedSignConvention: SpeedLimitSign?) {
+        when (speedSignConvention) {
+            SpeedLimitSign.MUTCD -> {
+                speedInfoMutcdLayout.isVisible =
+                    speedInfoOptions.renderWithSpeedSign != SpeedLimitSign.VIENNA
+                speedInfoViennaLayout.isVisible =
+                    !speedInfoMutcdLayout.isVisible
+            }
+            SpeedLimitSign.VIENNA -> {
+                speedInfoViennaLayout.isVisible =
+                    speedInfoOptions.renderWithSpeedSign != SpeedLimitSign.MUTCD
+                speedInfoMutcdLayout.isVisible =
+                    !speedInfoViennaLayout.isVisible
+            }
+            // TODO: remove the null check after NN-305 is available
+            null -> {
+                when (speedInfoOptions.renderWithSpeedSign) {
+                    null -> {
+                        speedInfoMutcdLayout.isVisible = speedInfoMutcdLayout.isVisible
+                        speedInfoViennaLayout.isVisible = speedInfoViennaLayout.isVisible
+                    }
+                    SpeedLimitSign.MUTCD -> {
+                        speedInfoMutcdLayout.isVisible = true
+                        speedInfoViennaLayout.isVisible = false
+                    }
+                    SpeedLimitSign.VIENNA -> {
+                        speedInfoMutcdLayout.isVisible = false
+                        speedInfoViennaLayout.isVisible = true
+                    }
+                }
+            }
+        }
+    }
+
+    private fun renderSpeedUnit(speedUnit: SpeedUnit) {
+        when (speedUnit) {
+            SpeedUnit.MILES_PER_HOUR -> {
+                speedInfoUnitTextMutcd.text = "mph"
+            }
+            SpeedUnit.KILOMETERS_PER_HOUR -> {
+                speedInfoUnitTextMutcd.text = "km/h"
+            }
+            else -> {
+                // No-op
+            }
+        }
+    }
+
+    private fun updateView() {
+        speedInfoUnitTextMutcd.isVisible = speedInfoOptions.showUnit
+        speedInfoLegendTextMutcd.isVisible = speedInfoOptions.showLegend
+        when (speedInfoOptions.currentSpeedDirection) {
+            CurrentSpeedDirection.TOP -> {
+                renderCurrentSpeedToTop()
+            }
+            CurrentSpeedDirection.END -> {
+                renderCurrentSpeedToEnd()
+            }
+            CurrentSpeedDirection.START -> {
+                renderCurrentSpeedToStart()
+            }
+            CurrentSpeedDirection.BOTTOM -> {
+                renderCurrentSpeedToBottom()
+            }
+        }
+    }
```

Comment:
Yes that is expected. Except for `showSpeedWhenUnavailable` and `renderWithSpeedSign` all other options should take effect immediately, while these 2 options will only be applied to next render call. There are other variables such as `SpeedInfo.speedSign` which is required to decide whether to show/hide MUTCD/VIENNA layout when `showSpeedWhenUnavailable=true`. We cannot just add a line 
`speedInfoMutcdLayout.isVisible = showSpeedWhenUnavailable` in this function since it does not have access to `speedInfo.speedSign`.